
	Cygwin DLL Makefile fixes.
	* target.make [cygwin]: Use '-Wl,--enable-auto-image-base' instead
	of manually rebasing.  Define DLL_PREFIX to match Cygwin conventions.
	* Instance/framework.make: Use DLL_PREFIX.
	* Instance/library.make: Ditto.

--- origsrc/gnustep-make-2.2.0/Instance/framework.make	2009-03-19 11:59:53.000000000 -0500
+++ src/gnustep-make-2.2.0/Instance/framework.make	2009-08-02 22:30:32.778992900 -0500
@@ -285,7 +285,7 @@ SONAME_FRAMEWORK_FILE  = $(FRAMEWORK_LIB
 # the INTERFACE_VERSION in the DLL library name.  Applications are
 # linked explicitly to this INTERFACE_VERSION of the library; this
 # works exactly in the same way as under Unix.
-LIB_LINK_DLL_FILE    = $(GNUSTEP_INSTANCE)-$(subst .,_,$(INTERFACE_VERSION))$(DLL_LIBEXT)
+LIB_LINK_DLL_FILE    = $(DLL_PREFIX)$(GNUSTEP_INSTANCE)-$(subst .,_,$(INTERFACE_VERSION))$(DLL_LIBEXT)
 
 FRAMEWORK_OBJ_EXT = $(DLL_LIBEXT)
 endif # BUILD_DLL
--- origsrc/gnustep-make-2.2.0/Instance/library.make	2009-03-19 11:59:53.000000000 -0500
+++ src/gnustep-make-2.2.0/Instance/library.make	2009-08-02 01:54:54.443694600 -0500
@@ -218,7 +218,7 @@ SONAME_LIBRARY_FILE  = $(LIBRARY_FILE)
 # the INTERFACE_VERSION in the DLL library name.  Applications are
 # linked explicitly to this INTERFACE_VERSION of the library; this
 # works exactly in the same way as under Unix.
-LIB_LINK_DLL_FILE    = $(LIBRARY_NAME_WITHOUT_LIB)-$(subst .,_,$(INTERFACE_VERSION))$(DLL_LIBEXT)
+LIB_LINK_DLL_FILE    = $(DLL_PREFIX)$(LIBRARY_NAME_WITHOUT_LIB)-$(subst .,_,$(INTERFACE_VERSION))$(DLL_LIBEXT)
 endif # BUILD_DLL
 
 else # following code for static libs
--- origsrc/gnustep-make-2.2.0/target.make	2009-03-24 18:08:59.000000000 -0500
+++ src/gnustep-make-2.2.0/target.make	2009-08-02 01:57:58.270253400 -0500
@@ -931,6 +931,7 @@ SHARED_LIB_LINK_CMD     = \
 	-Wl,--out-implib=$(LIB_LINK_OBJ_DIR)/$(LIB_LINK_VERSION_FILE) \
 	-Wl,--export-all-symbols \
 	-Wl,--enable-auto-import \
+	-Wl,--enable-auto-image-base \
 	-Wl,--whole-archive $(OBJ_FILES_TO_LINK) $(ALL_LDFLAGS) \
 	-Wl,--no-whole-archive $(INTERNAL_LIBRARIES_DEPEND_UPON) $(TARGET_SYSTEM_LIBS)\
 	$(SHARED_LD_POSTFLAGS)
@@ -942,9 +943,8 @@ SHARED_LIBEXT	 = .dll.a
 BUILD_DLL	 = yes
 CYGWIN_DLL_SUPPORT  = yes
 #SHARED_LIBEXT	 = .a
+DLL_PREFIX	 = cyg
 DLL_LIBEXT	 = .dll
-REBASE 		 = rebase 
-REBASE_FLAGS = -d -b 0x68000000 -o 0x10000
 CYGWIN_LD_FLAGS = -Wl,--export-all-symbols -Wl,--enable-auto-import
 #SHARED_CFLAGS	 += 
 
@@ -955,14 +955,14 @@ HAVE_BUNDLES   = yes
 BUNDLE_LD      = $(LD)
 BUNDLE_LDFLAGS += -shared -Wl,--export-all-symbols \
 	-Wl,--enable-auto-import \
+	-Wl,--enable-auto-image-base \
 	-Wl,--whole-archive
-BUNDLE_LIBFLAGS += -Wl,--no-whole-archiv
+BUNDLE_LIBFLAGS += -Wl,--no-whole-archive
 BUNDLE_LINK_CMD  = \
         $(BUNDLE_LD) $(BUNDLE_LDFLAGS) $(ALL_LDFLAGS) \
 	-o $(LDOUT)$(BUNDLE_FILE) \
 	$(OBJ_FILES_TO_LINK) \
-	$(BUNDLE_LIBFLAGS) $(ALL_LIB_DIRS) $(BUNDLE_LIBS); \
-	$(REBASE) $(REBASE_FLAGS) $(BUNDLE_FILE)
+	$(BUNDLE_LIBFLAGS) $(ALL_LIB_DIRS) $(BUNDLE_LIBS)
 endif
 
 # end Cygwin
